### DISCUSSION TITLE
@astrojs/image Picture Component CSS control

### DISCUSSION THREAD

**Discussion by Fish1** [withastro/roadmap#394] [Proposal]
I've run into a situation where I think I need control of the CSS of the ```<picture>``` tag within the Picture component.

Note: I am using tailwindcss with DaisyUI

I am getting these border on my card component because the ```<picture>``` tag produced by the Picture component is not taking up 100% of the width of the figure.

# Current Code & Result
```html
---
import { Picture } from '@astrojs/image/components';
import image2 from '../assets/image2.jpg';
---

<div class="card lg:w-1/2 md:w-full bg-base-100 shadow-xl image-full">
  <figure class="object-contain">
    <Picture
      src={image2}
      widths={[50, 1200]}
      aspectRatio={"16:9"}
      sizes="900px"
      fit="cover"
      formats={["jpeg"]}
      alt="A beautiful landscape"
    />
  </figure>
  <div class="card-body">
    <h3 class="card-title">Become the face of a better, safer tomorrow.</h3>
    <p>
      At ADT, you'll be entrusted to care for what matters most. Find a career you can take pride in.
    </p>
    <p>
      Explore Customer Service, Sales, Installation & Service and Corporate positions.
    </p>
    <div class="card-actions">
      <button class="btn btn-primary">Join</button>
    </div>
  </div>
</div>
```
![image](https://user-images.githubusercontent.com/13600696/202344910-12de5b69-ffa7-4787-a693-f12b8721f3ae.png)
![image](https://user-images.githubusercontent.com/13600696/202345066-6e34c484-822a-4dca-a2ef-a05cd83cb5f5.png)

# Hacky Fix with inspect element

I am able to get what I want by adding a max height to the figure, width 100% to the picture, and width 100% to the image.

![image](https://user-images.githubusercontent.com/13600696/202345418-140b344f-f5fd-4d9c-b22b-0c1111f92620.png)

I can get 2/3rds of the way there with the following code... adding a max height to the figure, and a 100% width to the image... But I cannot access the ```picture```
```html
 <figure class="object-contain max-h-96">
    <Picture
      src={image2}
      widths={[50, 1200]}
      aspectRatio={"16:9"}
      sizes="900px"
      fit="cover"
      formats={["jpeg"]}
      alt="A beautiful landscape"
      class="w-full"
    />
  </figure>
```

Going into ```Picture.astro``` this is the return of the Picture component.
```html
<picture>
	{sources.map((attrs) => <source {...attrs} {sizes} />)}
	<img {...image} {loading} {decoding} {alt} {...attrs} />
</picture>
```

Let me know if anyone has any thoughts on adding some controls for this... Or if there is something I could do better?

**Comment by dasDaniel** [comment]
I'm in the same boat...

It would be helpful to pass a props (or just the class) to the `<picture>` tag, not just the `<img>` tag

```html
<picture {...picture}>
	{sources.map((attrs) => <source {...attrs} {sizes} />)}
	<img {...image} {loading} {decoding} {alt} {...attrs} />
</picture>
```

I'm trying to place some images of various sizes in a gallery layout and would like to center them vertically and horizontally within their frame. 

The workaround I'm using is having a class for the parent tag and then set the styles with 

```css
.gallery picture {
  @apply flex items-center w-full mx-auto
}
```
